src/uu/tr/src/operation.rs

```diff
@@ -0,0 +1,501 @@
+use crate::unicode_table;
+use nom::{
+    branch::alt,
+    bytes::complete::{tag, take_while1},
+    character::complete::{anychar, one_of},
+    combinator::{map_opt, recognize},
+    multi::{many0, many_m_n},
+    sequence::{preceded, separated_pair, tuple},
+    IResult,
+};
+use std::{
+    collections::HashMap,
+    fmt::Debug,
+    io::{BufRead, Write},
+};
+
+static SPACES: &'static [char] = &[
+    unicode_table::HT,
+    unicode_table::LF,
+    unicode_table::VT,
+    unicode_table::FF,
+    unicode_table::CR,
+    unicode_table::SPACE,
+];
+static BLANK: &'static [char] = &[unicode_table::SPACE, unicode_table::HT];
+
+pub enum Sequence {
+    Char(char),
+    CharRange(Box<dyn Iterator<Item = char>>),
+    CharStar(char),
+}
+
+impl Sequence {
+    pub fn parse_set_string(input: &str) -> Vec<Sequence> {
+        many0(alt((
+            alt((Sequence::parse_octal, Sequence::parse_backslash)),
+            alt((
+                Sequence::parse_char_range,
+                Sequence::parse_char_star,
+                Sequence::parse_char_repeat,
+            )),
+            alt((
+                Sequence::parse_alnum,
+                Sequence::parse_alpha,
+                Sequence::parse_blank,
+                Sequence::parse_control,
+                Sequence::parse_digit,
+                Sequence::parse_graph,
+                Sequence::parse_lower,
+                Sequence::parse_print,
+                Sequence::parse_punct,
+                Sequence::parse_space,
+                Sequence::parse_space,
```

Comment:
I'm guessing we don't need 2 of these :)

Reply:
Yep xd